django/utils/autoreload.py

```diff
@@ -207,12 +207,27 @@ def get_child_arguments():
     on reloading.
     """
     import django.__main__
+    django_main_path = Path(django.__main__.__file__)
+    py_script = Path(sys.argv[0])
 
     args = [sys.executable] + ['-W%s' % o for o in sys.warnoptions]
-    if sys.argv[0] == django.__main__.__file__:
+    if py_script == django_main_path:
         # The server was started with `python -m django runserver`.
         args += ['-m', 'django']
         args += sys.argv[1:]
+    elif not py_script.exists():
+        # sys.argv[0] can not exist for several reasons on Windows.
+        # It may exist with a .exe extension or have a -script.py suffix
+        exe_entrypoint = py_script.with_suffix('.exe')
+        script_entrypoint = py_script.with_name('{}-script.py'.format(py_script.name))
+        if exe_entrypoint.exists():
+            # Should be executed directly, ignoring sys.executable
+            return [exe_entrypoint] + sys.argv[1:]
+        elif script_entrypoint.exists():
+            # Should be executed as usual
+            return args + [script_entrypoint] + sys.argv[1:]
+        else:
+            raise RuntimeError("Script {} does not exist.".format(py_script))
```

Comment:
`"Script {} does not exist.".format(py_script)` ->
`'Script %s does not exist." % py_script`